lib/Common/Memory/RecyclerWriteBarrierManager.cpp

```diff
@@ -66,6 +66,16 @@ X64WriteBarrierCardTableManager::OnThreadInit()
     ::GetCurrentThreadStackLimits(&stackEnd, &stackBase);
 #endif
 
+#ifdef X64_WB_DIAG
+    this->_stackbase = (char*)stackBase;
+    this->_stacklimit = (char*)stackEnd;
+#endif
+
+    // on Windows server 2012 stack limit can expand with process running, and causes
```

Comment:
I only hit this on CI build which is running windows server 2012. it might happen in win10 as well but I didn't get a repro on my machine for running 1000 times of the test

---
In reply to: [90075652](https://github.com/Microsoft/ChakraCore/pull/2105#discussion_r90075652) [](ancestors = 90075652)